app/gui/src/dashboard/components/AriaComponents/Form/components/Cancel.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * @file Cancel button for forms.
+ * Manages the form state and displays a loading spinner when the form is submitting.
+ */
+import type { JSX } from 'react'
+
+import { Form, useDialogContext, type ButtonProps } from '#/components/AriaComponents'
+import { useText } from '#/providers/TextProvider'
+import type { FormInstance } from './types'
+
+/**
+ * Additional props for the Cancel component.
+ */
+interface CancelButtonBaseProps {
+  readonly variant?: ButtonProps['variant']
+  /**
+   * Connects the submit button to a form.
+   * If not provided, the button will use the nearest form context.
+   *
+   * This field is helpful when you need to use the submit button outside of the form.
+   */
+  // We do not need to know the form fields.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  readonly form?: FormInstance<any>
+}
+
+/**
+ * Props for the Cancel component.
+ */
+export type CancelProps = CancelButtonBaseProps &
+  Omit<ButtonProps, 'formnovalidate' | 'href' | 'variant'>
+
+/**
+ * Cancel button for forms.
+ * Manages the form state and displays a loading spinner when the form is submitting.
+ */
+export function Cancel(props: CancelProps): JSX.Element {
```

Comment:
I don't see any relations to the `Form` at all. Could you elaborate please?

Reply:
well, it's still a `Form.Submit` button. although maybe it doesn't need to be? should it just be a regular button instead?

also note that *usually* i think this will be used inside a form anyway - we already have a separate `CloseButton` for dialogs themselves, right?

Reply:
`CloseButton` AFAIR is opinionated in terms of styling and supposed to be internal for dialogs.

> well, it's still a Form.Submit button. although maybe it doesn't need to be? should it just be a regular button instead?

I think yes, nothing's stopping us from using just a regular button + `useDialogContext`

> also note that usually i think this will be used inside a form anyway

Maybe yes, but not always. Even though the `Form` has integration with the `Dialog`, it doesn't mean that they should be always used together. It's just an implementation of the native behavior of the `<modal> + <form method="dialog">` pattern which I see useful in general: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#html-only_dialog